vivo-project/Vitro
# Decouple installation and deployment

**[VIVO GitHub issue](https://github.com/vivo-project/VIVO/issues/3034)**
[VIVO PR](https://github.com/vivo-project/VIVO/pull/4017)
# What does this pull request do?
Decoupled deployment and installation processes
Changed delivery of home directory: package it's contents as part of war artifact, update default files on home directory except rdf subdirectory on startup. 
In case tomcat directory is not provided artifact will not be copied into tomcat webapps directory

# What's new?
Changed deployment: war artifact is copied into webapp directory, not unpacked
Home directory contents are copied into war artifact
On startup path to home directory is being created if not already exists.
RDFFilesLoader and FileGraphSetup were adjusted to read home/rdf files from tomcat webapp application directory

# How should this be tested?
Test update of already installed instance and installation of new instance. 
- Verify that mvn install -s example-settings.xml work as before
- Remove tomcat directory from xml file, copy created war file manually to tomcat webapp directory, that should result in deployment of application.


# Additional Notes:
This change require documentation to be updated

# Interested parties
@VIVO-project/vivo-committers

# Reviewers' expertise

Candidates for reviewing this PR should have some of the following expertises:
1. Java
2. Maven
# Reviewers' report template
**Please update the following template which should be used by reviewers.**
## General comment
A reviewer should provide here comments and suggestions for requested changes if any.
## Testing
A reviewer should briefly describe here how it was tested
## Code reviewing
A reviewer should briefly describe here which part was code reviewed
